contracts/vaults/balancer/internal/pool/TwoTokenPoolUtils.sol

```diff
@@ -289,17 +289,21 @@ library TwoTokenPoolUtils {
         uint256 bptThreshold = strategyContext.vaultSettings._bptThreshold(
             poolContext.basePool.pool.totalSupply()
         );
-        uint256 bptHeldAfterJoin = strategyContext.totalBPTHeld + bptMinted;
+        uint256 bptHeldAfterJoin = strategyContext.vaultState.totalBPTHeld + bptMinted;
         if (bptHeldAfterJoin > bptThreshold)
             revert Errors.BalancerPoolShareTooHigh(bptHeldAfterJoin, bptThreshold);
 
         // Transfer token to Aura protocol for boosted staking
         bool success = stakingContext.auraBooster.deposit(stakingContext.auraPoolId, bptMinted, true); // stake = true
         if (!success) revert Errors.StakeFailed();
+
+        strategyContext.vaultState.totalBPTHeld += bptMinted;
+        strategyContext.vaultState.setStrategyVaultState(); 
```

Comment:
Updating totalBPTHeld here doesn't work @weitianjie2000 because it throws off the calculation in _convertBPTClaimToStrategyTokens. For that calculation to return the correct answer it must be done before you update totalBPTHeld instead of after.

Reply:
ok, I am moving the calculation back to where it was. I was gonna have it cover both deposit and reinvest rewards, but it seems like we'll need to update totalBPTHeld in both places.